src/handlers/autolabel.rs

```diff
@@ -69,11 +89,19 @@ pub(super) async fn parse_input(
                         name: label.to_owned(),
                     });
                 }
-                if cfg.new_pr && event.action == IssuesAction::Opened {
-                    autolabels.push(Label {
-                        name: label.to_owned(),
-                    });
-                }
+            }
+
+            // Treat the following situations as a "new PR":
+            // 1) New PRs opened as non-draft
+            // 2) PRs opened as draft that are marked as "ready for review" for the first time.
+            let is_new_non_draft_pr = event.action == IssuesAction::Opened && !event.issue.draft;
+            let is_first_time_ready_for_review =
+                event.action == IssuesAction::ReadyForReview && !state.data.new_pr_labels_applied;
+            if cfg.new_pr && (is_new_non_draft_pr || is_first_time_ready_for_review) {
+                autolabels.push(Label {
+                    name: label.to_owned(),
+                });
+                state.data.new_pr_labels_applied = true;
             }
```

Comment:
This should have stayed in the condition above, as we are putting new_pr labels on issues instead of only PRs.

I knew this was iffy, but couldn't prove it, thks @jieyouxu.

Reply:
Opened https://github.com/rust-lang/triagebot/pull/1983 to fix it.